Bugfix, baseuri in rdfskeleton is needed in the schema

Shamanou opened the pull request:
to load a rdf skeleton before you have saved you skeleton you need the baseuri to present in the schema when you call the openrefine backend.

Shamanou committed:
Bugfix, baseuri in rdfskeleton is needed in the schema

to load a rdf skeleton before you have saved you skeleton you need the baseuri to present in the schema when you call the openrefine backend.

kburger commented on extensions/fairifier-rdf-extension/scripts/rdf-skeleton-list.js:
```diff
@@ -27,10 +28,12 @@ RdfSkeletonListDialog.prototype._constructFooter = function(footer) {
                 project: self._project,
-                schema: JSON.stringify(self._schema),
-                projectId : listData[$( "select :selected" ).attr('id')].project
+                schema : self._schema,
```
Superfluous space before the colon?

kburger commented on extensions/fairifier-rdf-extension/scripts/rdf-skeleton-list.js:
```diff
@@ -27,10 +28,12 @@ RdfSkeletonListDialog.prototype._constructFooter = function(footer) {
-                schema: JSON.stringify(self._schema),
-                projectId : listData[$( "select :selected" ).attr('id')].project
+                schema : self._schema,
+                projectId : listData[$( "select :selected" ).attr('id')].project,
```
Trailing comma at the end?

kburger reviewed (approved):
Two minor comments, non breaking though. Looks like a simple and elegant solution. 👍

Shamanou committed:
Updated error reporting from backend to frontend

- main changes are in httputils, which propagate to other classes, i chose this router because we where already altering the files that where affected, the main OR code is uharmed, all tests (gui and backend) where passed, the change is to return the full http response object

- Added static references to httpheaders and mime types

- improved java style

- added more explicit exceptions

- removed duplicate httputils file

Shamanou committed:
Updated with PR comments, remove unneeded characters

Shamanou committed:
Fixed traves errors by making the addstanbolservice command compatible with new httputils

Also updated code style and migrated from jena to rdf4j

Shamanou committed:
Added new build.xml

kburger commented on extensions/fairifier-rdf-extension/dtls/fairifier/GetFairDataPointInfoCommand.java:
```diff
@@ -1,129 +1,160 @@
-public class GetFairDataPointInfoCommand extends Command{
+public class GetFairDataPointInfoCommand extends Command {
+
     private SimpleValueFactory f;
```
Not something from this commit, but this can be a constant. Right now it's initialized every time `doPost` is invoked.

kburger commented on extensions/fairifier-rdf-extension/dtls/fairifier/PostFairDataToFairDataPoint.java:
```diff
@@ -137,13 +138,23 @@ public void doPost(HttpServletRequest req, HttpServletResponse res)
+            respondException(res, ex);
+        } catch (IllegalAccessException ex) {
+            respondException(res, ex);
+        } catch (HttpException ex) {
```
How is this (the separate exceptions that are caught) handled in the front-end? Different exceptions have different causes and might require different actions from the user. As it is now, this could also be a multi-catch.

kburger commented on extensions/fairifier-rdf-extension/dtls/fairifier/Resource.java:
```diff
@@ -1,27 +1,32 @@
-    protected boolean hasModel(){
+
+    protected boolean hasModel() {
         if (this.fairData == null) return false;
```
[Missing curly brackets for an inline-if.](https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used)

kburger commented on extensions/fairifier-rdf-extension/deri/grefine/reconcile/commands/AddStanbolServiceCommand.java:
```diff
@@ -11,101 +12,121 @@
  * @author Sergio Fernandez <sergio.fernandez@salzburgresearch.at>
  *
  */
 public class AddStanbolServiceCommand extends Command {
```
Is this service being used in the reconciliation we normally do?

Shamanou replied:
yes

kburger reviewed (approved):
I might have missed some things here and there due to the diff-jungle of the formatting changes. Some minor comments, but for the rest I'm OK with this.

Shamanou committed:
Minor style update in resource file
